sdks/go/pkg/beam/core/graph/fn.go

```diff
@@ -1003,6 +1003,14 @@ func validateSdfSigTypes(fn *Fn, num int) error {
 					"Ensure that all restrictions in an SDF are the same type.",
 					truncateRestrictionName, 0, method.Ret[0].T, restrictionT, createInitialRestrictionName)
 			}
+			processFn := fn.methods[processElementName]
+			if len(processFn.Ret) > 0 && processFn.Ret[len(processFn.Ret)-1].T != reflect.TypeOf((*sdf.ProcessContinuation)(nil)).Elem() {
+				err := errors.Errorf("mismatched output type in method %v, return %v: got: %v, want: %v",
+					processElementName, len(processFn.Ret)-1, method.Ret[len(processFn.Ret)-1].T, reflect.TypeOf((*sdf.ProcessContinuation)(nil)).Elem())
+				return errors.SetTopLevelMsgf(err, "Mismatched output type in method %v, "+
+					"return value at index %v. Got: %v, Want: %v (from method %v).",
+					processElementName, len(processFn.Ret)-1, method.Ret[len(processFn.Ret)-1].T, reflect.TypeOf((*sdf.ProcessContinuation)(nil)).Elem(), truncateRestrictionName)
+			}
```

Comment:
Since processFn is a `funcx.Fn` you can check if ProcessContinuations are part of the signature by calling:
`i, exists := processFn.ProcessContinuation()`

https://github.com/apache/beam/blob/6de01035994cb5aa5a62c6f2ab47082dc06aaedd/sdks/go/pkg/beam/core/funcx/fn.go#L324

As written, you're only checking the last paraemter, which might be an `error` instead.

Reply:
oh, got it. Done.